R/4_standard_error.R

```diff
@@ -48,21 +56,42 @@ standard_error <- function(model, ...) {
 
 # Default methods ---------------------------------------------------------
 
-
-## TODO use "robust" argument instead of method = "robust"
-
 #' @rdname standard_error
 #' @export
-standard_error.default <- function(model, method = NULL, verbose = TRUE, ...) {
-  if (!is.null(method)) {
-    method <- tolower(method)
-  } else {
-    method <- "wald"
+standard_error.default <- function(model,
+                                   vcov = NULL,
+                                   vcov_args = NULL,
+                                   verbose = TRUE,
+                                   ...) {
+
+  dots <- list(...)
+
+  se <- NULL
+
+  # vcov: matrix
+  if (is.matrix(vcov)) {
+    se <- sqrt(diag(vcov))
   }
 
-  if (method == "robust") {
-    standard_error_robust(model, ...)
-  } else {
+  # vcov: function which returns a matrix
+  if (is.function(vcov)) {
+    args <- c(list(model), vcov_args, dots)
+    se <- tryCatch(sqrt(diag(do.call("vcov", args))),
+                   error = function(x) NULL)
+  }
+
+  # vcov: character (with backward compatibility for `robust = TRUE`)
+  if (is.character(vcov) || isTRUE(dots[["robust"]])) {
```

Comment:
Do we want to give a deprecation warning?

Reply:
I will include a warning in `.get_vcov()` since everything flows through there.